tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -3134,11 +3134,33 @@ void GetTensorDimsWithProtoShape(const Tensor& tensor, nvinfer1::Dims* dims) {
   }
 }
 
+template <typename Bounds, typename Input>
+constexpr bool IsIntegerInBounds(const Input& inp) {
+  static_assert(std::is_integral<Input>::value,
+                "This function is only implemented for integral types.");
+  return (sizeof(Bounds) >= sizeof(Input))
+             // Since Input is smaller, it will always be in range of Bounds.
+             ? true
+             // Since Input is larger, we cast all values to that.
+             : (inp >=
+                    static_cast<Input>(std::numeric_limits<Bounds>::lowest()) &&
+                inp <= static_cast<Input>(std::numeric_limits<Bounds>::max()));
+}
+
 template <DataType dtype>
-void CopyToTrtInt32Array(const Tensor& tensor, int32* dst) {
+Status CopyToTrtInt32Array(const Tensor& tensor, int32* dst) {
   typedef typename EnumToDataType<dtype>::Type CType;
   const CType* src = tensor.flat<CType>().data();
+  for (int i = 0; i < tensor.NumElements(); ++i) {
```

Comment:
Please add a fast-path: `if (!std::is_same<int32, CType>::value) { ...do the check; }`

Reply:
I think the condition in `isIntegerInBounds`:
```
if (std::numeric_limits<Input>::lowest() >= std::numeric_limits<Bounds>::lowest()
      && std::numeric_limits<Input>::max() <= std::numeric_limits<Bounds>::max())
```
would be evaluated at compile-time, so the compiler can completely optimize away the check in cases where it's unnecessary.

Reply:
Even in that case, it still need to do the copy below `dst[i] = static_cast<int32>(src[i]);`. Can we get rid of that?

Also, to make the compiler's life easier please move the bound check outside the function and the for loop.

Reply:
Removed the redundant copy. We still need the check to be inside the loop, since we need to check each individual value in the tensor to make sure it's in bounds.